pennylane/math/single_dispatch.py

```diff
@@ -33,6 +33,11 @@ def _i(name):
     return import_module(name)
 
 
+# ------------------------------- Builtins -------------------------------- #
+
+ar.register_function("builtins", "ndim", lambda x: np.ndim(np.array(x)))
+ar.register_function("builtins", "shape", lambda x: np.array(x).shape)
+

```

Comment:
Added as part of the recent update to autoray, this should be removed once master is updated with the latest fix for it. Until then, it is kept here so that CI can pass.